### PR TITLE
fix(seg): keep RF-DETR mask loss graph connected

### PR DESCRIPTION
**Pull request by imagra93**
## Problem

RF-DETR segmentation training with `static_graph=True` can crash when a rank sees a batch with zero Hungarian matches, such as an all-background batch.

In the factored mask-head branch, `loss_masks` returned a disconnected empty tensor for the no-match case. That tensor has no `grad_fn`, so the mask-head tensors do not participate in backward on that rank, which violates DDP's static-graph requirement.

Background/unlabeled images are valid negative samples and should remain in the training dataset.

Addresses the DDP unlabeled-image crash in #291.

## Solution

Return zero mask losses routed through the mask-head tensors instead of a disconnected empty tensor:

```python
zero = spatial_features.sum() * 0.0 + query_features.sum() * 0.0 + bias.sum() * 0.0
return {"loss_mask_ce": zero, "loss_mask_dice": zero}
```

The loss value is still zero, but `spatial_features`, `query_features`, and `bias` remain in the backward graph and receive zero gradients. This keeps DDP's reducer stable without dropping background images.

A regression test covers the no-match mask-loss path and asserts that backward keeps all mask-head tensors connected with zero gradients.

### CODE REVIEW

**Comment by imagra93** [comment]
@codex review

**Comment by EHxuban11** [comment]
I'm concerned that this PR is dropping the unlabelled images basically (often called background images)d, but background images are useful in order to avoid the model to detect false positives when there isn't an object in the image.

**Comment by imagra93** [comment]
Correct: I keep your unlabeled images. The DDP crash is gone.

A selective filter (e.g. max_unlabeled_ratio=0.X) could be a useful future option for datasets where the unlabeled fraction is so large it dominates the sampler and starves the model of positive examples. But the right threshold is dataset-dependent and we don't have enough testing to pick a sensible default yet, so we're leaving that for a follow-up if there's demand.

**Comment by EHxuban11** [comment]
Thank you @imagra93 for your contribution. This is mergeable as it is but i'll add a small test first.
